bitnami/jupyterhub/templates/NOTES.txt

```diff
@@ -0,0 +1,64 @@
+
+   export CLUSTER_IP=$(minikube ip) # On Minikube. Use: `kubectl cluster-info` on others K8s clusters
+   export HOSTNAME=$(kubectl get ingress --namespace {{ .Release.Namespace }} {{ template "common.names.fullname" . }} -o jsonpath='{.spec.rules[0].host}')
+   echo "JupyterHub URL: http://$HOSTNAME/"
+   echo "$CLUSTER_IP  $HOSTNAME" | sudo tee -a /etc/hosts
+
+{{- else }}
+
+1. Get the JupyterHub URL by running:
+
+{{- if eq .Values.proxy.service.public.type "NodePort" }}
+
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "common.names.fullname" . }}-proxy-public)
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo "JupyterHub URL: http://$NODE_IP:$NODE_PORT/"
+
+{{- else if eq .Values.proxy.service.public.type "LoadBalancer" }}
+
+** Please ensure an external IP is associated to the {{ template "common.names.fullname" . }}-proxy-public service before proceeding **
+** Watch the status using: kubectl get svc --namespace {{ .Release.Namespace }} -w {{ template "common.names.fullname" . }}-proxy-public **
+
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "common.names.fullname" . }}-proxy-public --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+
+{{- $port:=.Values.proxy.service.public.port | toString }}
+  echo "JupyterHub URL: http://$SERVICE_IP{{- if ne $port "80" }}:{{ .Values.proxy.service.public.port }}{{ end }}/"
+{{- else if eq .Values.proxy.service.public.type "ClusterIP" }}
+
+  echo "JupyterHub URL: http://127.0.0.1:8080/"
+  kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ template "common.names.fullname" . }}-proxy-public 8080:{{ .Values.proxy.service.public.port }}
+
+{{- end }}
+{{- end }}
+
+2. Login with the following admin user below to create a Notebook:
+
+  echo Admin user: {{ .Values.hub.adminUser }}
+  echo Password: $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "common.names.fullname" . }}-hub -o jsonpath="{.data['values\.yaml']}" | base64 --decode | awk -F: '/password/ {gsub(/[ \t]+/, "", $2);print $2}')
+
+{{- include "common.warnings.rollingTag" .Values.hub.image }}
+{{- include "common.warnings.rollingTag" .Values.proxy.image }}
+{{- include "common.warnings.rollingTag" .Values.singleuser.image }}
+
+{{- $passwordValidationErrors := list -}}
+
+{{- if not .Values.hub.existingSecret -}}
+    {{- $secretName := printf "%s-hub" (include "common.names.fullname" .) -}}
+    {{- $requiredHubPassword := dict "valueKey" "hub.password" "secret" $secretName "field" "values.yaml" "context" $ -}}
+    {{- $requiredHubPasswordError := include "common.validations.values.single.empty" $requiredHubPassword -}}
+    {{- $passwordValidationErrors = append $passwordValidationErrors $requiredHubPasswordError -}}
+{{- end }}
```

Comment:
Correct me if I'm wrong. If a user sets a custom configuration using `hub.configuration` it can potentially change the authentication mechanism as it's explained in the "Authenticator" section of the **README.md**. In this case, this validation should be skipped.

Reply:
Yes, but I don't know how to properly detect if this has been overridden. The easiest thing I could do is to disable the validation always

Reply:
I think we should disable the validation if the `hub.configuration` parameter is set. As you mentioned, we cannot ensure that this property is overridden in the custom configuration but this is safer rather than blocking an upgrade with no reasons to do it.